docs/spark-procedures.md

```diff
@@ -462,16 +462,29 @@ will then treat these files as if they are part of the set of files  owned by Ic
 
 #### Usage
 
-| Argument Name | Required? | Type | Description |
-|---------------|-----------|------|-------------|
-| `table`       | ✔️  | string | Table which will have files added to|
-| `source_table`| ✔️  | string | Table where files should come from, paths are also possible in the form of \`file_format\`.\`path\` |
-| `partition_filter`  | ️   | map<string, string> | A map of partitions in the source table to import from |
+| Argument Name | Required? | Type | Description                                                                                                                                            |
+|---------------|-----------|------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `table`       | ✔️  | string | Table which will have files added to                                                                                                                   |
+| `source_table`| ✔️  | string | Table where files should come from, paths are also possible in the form of \`file_format\`.\`path\`                                                    |
+| `partition_filter`  | ️   | map<string, string> | A map of partitions in the source table to import from                                                                                                 |
+| `check_duplicate_files`  | ️   | boolean | When true, will throw a exception if files added will result in duplicate (on by default, it's checking against files path in entries metadata table). |
 
 Warning : Schema is not validated, adding files with different schema to the Iceberg table will cause issues.
 
 Warning : Files added by this method can be physically deleted by Iceberg operations
 
+Warning : SQL delete followed by this add_files procedure immediately might not add files back due to deleted files are still tracked in manifest entry with status = 2: DELETED
```

Comment:
This part is a bit confusing.   Debating whether it makes sense or not to add it, whether its a niche use case that may just confuse the user.

But trying to understand the scope, do you know if it breaks  if the current snapshot has that entry as DELETED, or if any of the snapshots has that entry as DELETED?

If latter, maybe:
```
Files previously deleted from a table cannot be restored by this procedure while they are still tracked by the table metadata. 
```

Reply:
Actually if its just because of DELETED in current snapshot, is it because of check_duplicate_files?  If so, maybe we should just make a patch that selects out DELETED in that check.  Havent looked at the code yet.

Reply:
thank you @szehon-ho , I cut #6889 instead to fix the problem instead to callout as warning here. Will drop this line